packages/client/modules/userDashboard/components/GcalModal/GcalModal.tsx

```diff
@@ -0,0 +1,171 @@
+    color: PALETTE.SLATE_600
+  }
+})
+
+const StyledDialogContainer = styled(DialogContainer)({
+  width: 'auto'
+})
+
+const CloseIcon = styled(Close)({
+  color: PALETTE.SLATE_600,
+  cursor: 'pointer',
+  '&:hover': {
+    opacity: 0.5
+  }
+})
+
+const StyledCloseButton = styled(PlainButton)({
+  height: 24,
+  marginLeft: 'auto'
+})
+
+const ErrorMessage = styled(StyledError)({
+  textAlign: 'left',
+  paddingBottom: 8
+})
+
+const validateTitle = (title: string) => {
+  return new Legitity(title).trim().min(2, `C’mon, you call that a title?`)
+}
+
+interface Props {
+  handleCreateGcalEvent: (CreateGcalEventInput: CreateGcalEventInput) => void
+  closeModal: () => void
+}
+
+const GcalModal = (props: Props) => {
+  const {handleCreateGcalEvent, closeModal} = props
+
+  const startOfNextHour = dayjs().add(1, 'hour').startOf('hour')
+  const endOfNextHour = dayjs().add(2, 'hour').startOf('hour')
+  const [start, setStart] = useState(startOfNextHour)
+  const [end, setEnd] = useState(endOfNextHour)
+  const [inviteTeam, setInviteTeam] = useState(true)
+
+  const {fields, onChange} = useForm({
+    title: {
+      getDefault: () => ''
+    },
+    description: {
+      getDefault: () => ''
+    }
+  })
+  const titleErr = fields.title.error
+
+  const handleClick = () => {
+    const title = fields.title.value
+    const titleRes = validateTitle(title)
+    if (titleRes.error) {
+      fields.title.setError(titleRes.error)
+      return
+    }
+    const startTimestamp = start.unix()
+    const endTimestamp = end.unix()
+    const description = fields.description.value
+    const timeZone = Intl.DateTimeFormat().resolvedOptions().timeZone
+    const input = {
+      title,
+      description,
+      startTimestamp,
+      endTimestamp,
+      inviteTeam,
+      timeZone
+    }
+    handleCreateGcalEvent(input)
+  }
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    if (titleErr) {
+      fields.title.setError('')
+    }
+    onChange(e)
+  }
+
+  return (
+    <StyledDialogContainer>
+      <DialogTitle>
+        {'Schedule Your Meeting'}
+        <StyledCloseButton onClick={closeModal}>
+          <CloseIcon />
+        </StyledCloseButton>
+      </DialogTitle>
+      <DialogContent>
+        {
+          'Tell us when you want to meet and we’ll create a Google Calendar invite with a Parabol link'
+        }
+        <div className='space-y-1 pt-4'>
+          <div>
+            <StyledInput
+              autoFocus
+              maxLength={100}
+              defaultValue={fields.title.value}
+              onChange={handleChange}
+              name='title'
+              placeholder='Enter the name of your meeting'
+            />
+            {titleErr && <ErrorMessage>{titleErr}</ErrorMessage>}
+          </div>
+          <StyledInput
+            maxLength={100}
+            name='description'
+            onChange={onChange}
+            placeholder='Enter your meeting description (optional)'
```

Comment:
+1 I would prefer if we'd have 1 meeting name dialog which is also used for the actual meeting name. I think this needs a UI iteration going forward